phoenix-core/src/main/java/org/apache/phoenix/query/ConnectionQueryServicesImpl.java

```diff
@@ -4294,11 +4295,9 @@ public boolean acquireUpgradeMutex(long currentServerSideTableTimestamp)
     public boolean writeMutexCell(String tenantId, String schemaName, String tableName,
             String columnName, String familyName) throws SQLException {
         try {
-            byte[] rowKey =
-                    columnName != null
-                            ? SchemaUtil.getColumnKey(tenantId, schemaName, tableName, columnName,
-                                familyName)
-                            : SchemaUtil.getTableKey(tenantId, schemaName, tableName);
+            byte[] rowKey = columnName != null ?
+                SchemaUtil.getColumnKey(tenantId, schemaName, tableName, columnName, familyName) :
+                SchemaUtil.getTableKey(tenantId, schemaName, tableName);
             // at this point the system mutex table should have been created or
             // an exception thrown
             byte[] sysMutexPhysicalTableNameBytes = getSysMutexPhysicalTableNameBytes();
```

Comment:
If I understand correctly what @ChinmaySKulkarni described in the ticket, this call will still result to an admin.tableExists call to check the existance of SYSTEM.MUTEX/SYSTEM:MUTEX and you didn't changed that.

Reply:
Instead We could try the Table sysMutexTable =getTable() call with one of them and catch HBase TableNotFoundException, in that case try with the other one.

Reply:
Here is the throw part: https://github.com/apache/phoenix/pull/920/files#diff-539e6818644fa837ebee4e4b61f4c570f75ac439437050c9a0459762ef8e385cR4373

Reply:
@virajjasani I think there is some confusion here. The aim of this Jira is to reduce the HBase calls to get the table. Currently, the call to `writeMutexCell()` calls `getSysMutexPhysicalTableNameBytes()` which does 1 or 2 HBase admin calls (tableExists()) and then we still do a getTable() call [here](https://github.com/apache/phoenix/blob/628fa0db239985b2972f51459540f7513d182885/phoenix-core/src/main/java/org/apache/phoenix/query/ConnectionQueryServicesImpl.java#L4305). 
The same happens for `deleteMutexCell()`. 

Instead of doing this, we can do 1 getTable call with `SYSTEM.MUTEX` and if that throws a TNFE, try again with `SYSTEM:MUTEX`.

Reply:
My bad for this misunderstanding. Addressed concerns, updated the PR.